### ISSUE TITLE
False-positive: Condition duplicated in another if/elseif branch

### ISSUE THREAD

**Issue by Debeet** [kalessil/phpinspectionsea#339]
```
if ($start < 0) {
   $start = iconv_strlen($s, $encoding) + $start;
   if ($start < 0) { // <-- This condition is duplicated in another if/elseif branch.
      $start = 0;
   }
}
```

In theory the inspection is not wrong. This condition _is_ duplicated but it gives impression that it's a mistake in the code when in reality `$start` can have a different value in the second condition.

**Comment by rentalhost** [comment]
1. You can rename the second variable.
2. The inspection could check if variable is overwrited after initial condition and "reset usage status" (ping @kalessil);

**Comment by funivan** [comment]
how `iconv_strlen` can return negative value ?

**Comment by Doqnach** [comment]
I also run into this in our codebase with some frequency (often in the form of an `$error` boolean or similar)

**Comment by kalessil** [comment]
Due to analysis workflow, I don't have a solution here - this getting reported all the time, but no chance to fix it in short term perspective.

Hence, with the 2.3.8-preview we introduced options for the inspection, where possible to skip reporting duplicate conditions (at all). Or just rename the inner variable, what also makes sense.

**Comment by kalessil** [comment]
Fixed #550, coming into 2.3.12
